feat: pass `docs_map` to Executor (#5366)
jina-ai/jina: docs/fundamentals/executor/executor-methods.md

- `parameters`: A Dict object that passes extra parameters to Executor functions.

> - `docs_matrix`:  This is one of the least common parameter to be used for an Executor. This is needed when an Executor is used inside a Flow to merge or reduce the output of more than one other Executor.

Reviewer: ```suggestion
- `docs_matrix`:  This is one of the least common parameters to be used for an Executor. It is passed when multiple parallel branches lead into the Executor, and `disable_reduce=True` is set. Each DocumentArray in the matrix is the output of one previous Executor.
```
This is explained in more detail below, but I would still put it here

> - `docs_map`:  This is one of the least common parameter to be used for an Executor. It has the same utility as `docs_matrix` but the information comes as a dict with previous Executor names as keys. If onl

Reviewer: ```suggestion
- `docs_map`:  This is one of the least common parameter to be used for an Executor. It has the same utility as `docs_matrix` but the information comes as a dict with previous Executor names as keys, and DocumentArrays as values.
```

> - `docs_map`:  This is one of the least common parameter to be used for an Executor. It has the same utility as `docs_matrix` but the information comes as a dict with previous Executor names as keys. If onl

Reviewer 2: ```suggestion
- `docs_map`:  This is also one of the least common parameter to be used for an Executor. It has the same utility as `docs_matrix` but the information comes as a dict with previous Executor names as keys.
```

> - `tracing_context`: Context needed if you want to add custom traces. Check {ref}`how to add custom traces in your Executor <instrumenting-executor>`

Reviewer: ```suggestion
- `tracing_context`: Context needed if you want to add custom traces. Check {ref}`how to add custom traces in your Executor <instrumenting-executor>`.
```

### Multiple DocumentArrays as input argument

You have seen that {class}`~jina.Executor` methods can receive multiple parameters.

`docs_matrix` and `docs_map` are only used in some special cases.

> One case is when an Executor receives messages from more than one upstream {class}`~jina.Executor` in the {class}`~jina.Flow`:

Reviewer 2: ```suggestion
One case is when an Executor receives messages from more than one incoming {class}`~jina.Executor` in the {class}`~jina.Flow`:
```
or maybe we say preceding

)

with f:
    returned_docs = f.post(on='/', inputs=Document())

print(f'Resulting documents {returned_docs[0].text}')
```

```shell
────────────────────────── 🎉 Flow is ready to serve! ──────────────────────────
╭────────────── 🔗 Endpoint ───────────────╮
│  ⛓     Protocol                    GRPC  │
│  🏠       Local           0.0.0.0:55761  │
│  🔒     Private     192.168.1.187:55761  │
│  🌍      Public    212.231.186.65:55761  │
╰──────────────────────────────────────────╯

MergeExec processing pairs of Documents "Exec1" and "Exec2"
Resulting documents Document merging from "Exec1" and "Exec2"
```

When merging Documents from more than one upstream Executor, sometimes you want to control which Documents come from which Executor.

> You can use the `docs_by_executor` argument to make sure that the Executor will receive the `docs_matrix` as a dictionary where the `key` will be the last Executor 

Reviewer: ```suggestion
You can use the `docs_by_executor` argument to make sure that the Executor will receive the `docs_matrix` as a dictionary where the key will be the last Executor 
```
